source/val/validate_memory.cpp

```diff
@@ -1410,7 +1410,60 @@ spv_result_t ValidatePtrAccessChain(ValidationState_t& _,
              << "VariablePointers or VariablePointersStorageBuffer";
     }
   }
-  return ValidateAccessChain(_, inst);
+
+  // Need to call first, will make sure Base is a valid ID
+  if (auto error = ValidateAccessChain(_, inst)) return error;
+
+  const auto base_id = inst->GetOperandAs<uint32_t>(2);
+  const auto base = _.FindDef(base_id);
+  const auto base_type = _.FindDef(base->type_id());
+  const auto base_type_storage_class = base_type->word(2);
+
+  if (_.HasCapability(SpvCapabilityShader) &&
+      (base_type_storage_class == SpvStorageClassUniform ||
+       base_type_storage_class == SpvStorageClassStorageBuffer ||
+       base_type_storage_class == SpvStorageClassPhysicalStorageBuffer ||
+       base_type_storage_class == SpvStorageClassPushConstant) &&
+      !_.HasDecoration(base_type->id(), SpvDecorationArrayStride)) {
+    return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << "OpPtrAccessChain must have a Base whose type is decorated "
+              "with ArrayStride";
+  }
+
+  if (spvIsVulkanEnv(_.context()->target_env)) {
+    if (base_type_storage_class == SpvStorageClassWorkgroup) {
+      if (!_.HasCapability(SpvCapabilityVariablePointers)) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << _.VkErrorID(4707)
+               << "OpPtrAccessChain Base operand pointing to Workgroup "
+                  "storage class must use VariablePointers capability";
+      }
+    } else if (base_type_storage_class == SpvStorageClassStorageBuffer) {
+      if (!_.features().variable_pointers) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << _.VkErrorID(4707)
+               << "OpPtrAccessChain Base operand pointing to StorageBuffer "
+                  "storage class must use VariablePointers or "
+                  "VariablePointersStorageBuffer capability";
+      }
+    } else if (base_type_storage_class ==
+               SpvStorageClassPhysicalStorageBuffer) {
+      if (_.addressing_model() != SpvAddressingModelPhysicalStorageBuffer64) {
```

Comment:
This feels like the wrong spot for this check. This probably belongs in the validation of OpTypePointer.

Reply:
So the VUID was for a `OpPtrAccessChain` who's `Base`'s is using `PhysicalStorageBuffer` so I guess not sure how I would be able to check that in `OpTypePointer`

Reply:
Well in the SPIR-V spec for the physical storage buffer rules it says:
> If the [addressing model](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Addressing_Model) is not PhysicalStorageBuffer64, then the PhysicalStorageBuffer [storage class](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Storage_Class) must not be used.

That's why I suggested OpTypePointer. It will catch more cases.

Reply:
Ok I see, I agree, I think the `VUID-StandaloneSpirv-Base-04707` probably could just be reworded, going to remove this check from this PR (and make a new PR with it in `OpTypePointer` and have more tests aimmed towards that) and going to remove the VUID label as it will probably get a new one after my purposed spec clarification